### DISCUSSION TITLE
How to generate output from USB device?

### DISCUSSION THREAD

**Discussion by bkaz4283** [hathach/tinyusb#765] [Q&A] [answered]
Hello,

I am trying to implement MIDI over USB in my product using TinyUSB. The "Getting Started" instructions say to "call tud_int_handler()" in my USB IRQ handler and continuously call tud_task() if I'm not using any RTOSes (I am not).

My device is an audio to MIDI converter so most of what it will be doing is generating MIDI commands to send to the host, I don't anticipate the host will have any tasks at all for the device to process (with the exception of enumeration when the device is first connected).  In this case, should I just call the tud_midi_n_send() function whenever I have a new MIDI command to send?  And do I really need to call the tud_task() function periodically?

Thanks!
-Brian

**Comment by hathach** [comment, marked as answer]
https://github.com/hathach/tinyusb/tree/master/examples/device/midi_test
